### PR TITLE
🦄 Reverse membership lookup

### LINKED ISSUE #2032: Reverse membership lookup with external resources
# Background

Often, I find someone on Discord or Github, and I may have seen them before on that platform, or their content indicates to me that they are quite involved in the project, however, I don't know who they are as a member. 

# Scope

There should be a way to do reverse lookups from such external platform handles, likely on the membership screen which already has handle based lookup.

# Dependencies

https://github.com/Joystream/pioneer/issues/1977

### PR DESCRIPTION
**Pull request by WRadoslaw**
resolves #2032

### CODE REVIEW

**Comment by thesan** [comment]
@WRadoslaw please merge `carthage` into this one. It has a lot more changes than it should

File: packages/ui/src/memberships/components/SocialMediaSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  TELEGRAM: 'Enter Username',
+  TWITTER: 'Enter Username',
+  EMAIL: 'Enter Email',
+}
+
+interface Props {
+  initialSocials?: Socials[]
+}
+
+export const SocialMediaSelector = ({ initialSocials }: Props) => {
```
Comment (thesan) [inline packages/ui/src/memberships/components/SocialMediaSelector.tsx:29]: Wasn't this file part of the other PR ?

File: packages/ui/src/memberships/components/MemberListFilters/MemberListFilters.tsx [hunk 2]
```diff
@@ -41,20 +49,55 @@ export const MemberListEmptyFilter: MemberListFilter = {
...
+  'Email',
+  'Discord',
+  'Irc',
+  'Telegram',
+  'Wechat',
+  'Whatsapp',
+  'Twitter',
+  'Membership',
+  'Youtube',
+]
```
Comment (thesan) [inline packages/ui/src/memberships/components/MemberListFilters/MemberListFilters.tsx:90]: Hyperlink is missing

**Review by thesan — CHANGES_REQUESTED** [review]
I finally got to this one! I just have a few CR and a question

File: packages/ui/src/memberships/components/SocialMediaSelector.tsx (hunk 1 above)
Comment (WRadoslaw) [reply packages/ui/src/memberships/components/SocialMediaSelector.tsx:29]: Yeah, it was. Idk why it didn't get removed on merge,

File: packages/ui/src/memberships/hooks/useMembers.ts [hunk 3]
```diff
@@ -50,13 +50,46 @@ type FilterGqlInput = Pick<
...
+  }
+  if (searchFilter === 'Membership') {
+    return { handle_contains: search }
+  }
+
+  if (searchFilter === 'Membership_ID') {
+    return { id_eq: search }
+  }
+
+  if (searchFilter)
```
Comment (thesan) [inline packages/ui/src/memberships/hooks/useMembers.ts:84]:
```suggestion
```

**Review by thesan — CHANGES_REQUESTED** [review]
LGTM except for just one if statement I believe you forgot

**Review by thesan — APPROVED** [review]
:100:

**Pull request merged**